### ISSUE TITLE
setAllToExcluded also sets fields not specified in insert to excluded when using INSERT .. SET syntax

### ISSUE THREAD

**Issue by saibotma** [jOOQ/jOOQ#14599]
### Expected behavior

Only sets fields specified in the insert statement to excluded.

```sql
insert into "mm"."test" ("id")
values (1)
on conflict ("id")
do update
set
  "id" = excluded."id"
```

### Actual behavior

Sets all fields of the inserted table to excluded.

```sql
insert into "mm"."test" ("id")
values (1)
on conflict ("id")
do update
set
  "id" = excluded."id",
  "value" = excluded."value"
```

### Steps to reproduce the problem

Example table:
```sql
CREATE TABLE "mm"."test"
(
    id    int PRIMARY KEY,
    value text
);
```

Example insert:
```kotlin
insertInto(MM_TEST)
    .set(MM_TEST.ID, 
    .onDuplicateKeyUpdate()
    .setAllToExcluded()
    .execute()
```

### jOOQ Version

3.17.7

### Database product and version

PostgreSQL 15.1 on x86_64-pc-linux-musl, compiled by gcc (Alpine 12.2.1_git20220924-r4) 12.2.1 20220924, 64-bit

### Java Version

openjdk version "17.0.5" 2022-10-18

### OS Version

MacOs Ventura 13.1 (22C65)

### JDBC driver name and version (include name if unofficial driver)

org.postgresql:postgresql:42.5.1

**Comment by lukaseder** [comment]
Thanks for your report. This looks like a flaw in the implementation of `INSERT .. SET`. You'll get the expected output for:

```java
insertInto(MM_TEST)
    .columns(MM_TEST.ID).values(1) // instead of .set(MM_TEST.ID, 1)
    .onDuplicateKeyUpdate()
    .setAllToExcluded()
    .execute()
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0 and 3.17.8 (https://github.com/jOOQ/jOOQ/issues/14600)
